frontend/app/assets/javascripts/spree/frontend/checkout/coupon-code.js.coffee

```diff
@@ -0,0 +1,34 @@
+Spree.onCouponCodeApply = (e) ->
+  coupon_code_field = $('#order_coupon_code')
+  coupon_code = $.trim(coupon_code_field.val())
+  return if coupon_code == ''
+
+  coupon_status = $("#coupon_status")
```

Comment:
I know that a bunch of this is being copied over but it would be a good opportunity to convert some of these variables into `camelCase`. The new stuff that you've added uses it 👍 so we might as well make everything in here a little nicer if we can.